src/reducers/walletReducer.js

```diff
@@ -34,24 +35,34 @@ export default function (state = initState, action) {
   switch (action.type) {
     // metamask
     case 'CHECK_METAMASK_CONNECTION_FULFILLED':
-      return update(state, { wallet: { metamask: { $set: action.payload } } })
+      return update(state, { wallet: { metamask: { $merge: action.payload } } })
     case 'UPDATE_METAMASK_ACCOUNTS':
       return update(state, { wallet: { metamask: { accounts: { $set: action.payload } } } })
     // ledger
     case 'CHECK_LEDGER_NANOS_CONNECTION_FULFILLED':
-      return update(state, { wallet: { ledger: { $set: action.payload } } })
+      return update(state, { wallet: { ledger: { $merge: action.payload } } })
+    case 'CHECK_LEDGER_NANOS_CONNECTION_PENDING':
+      return update(state, { wallet: { ledger: { $merge: { connected: false } } } })
     case 'CHECK_LEDGER_NANOS_CONNECTION_REJECTED':
       return update(state, { wallet: { ledger: { $set: {
         connected: false,
         network: null,
-        accounts: null
+        accounts: {}
       } } } })
     // escrow wallet actions
     case 'VERIFY_PASSWORD_FULFILLED':
     // store decrypted wallet
       return update(state, { escrowWallet: { decryptedWallet: { $set: action.payload } } })
     case 'CLEAR_DECRYPTED_WALLET':
       return update(state, { escrowWallet: { decryptedWallet: { $set: null } } })
+    case 'SYNC_LEDGER_ACCOUNT_INFO_FULFILLED':
+      return update(state, { wallet: { ledger: { crypto: { $merge: action.payload } } } })
+    case REHYDRATE:
```

Comment:
why do we need this step?

Reply:
To initiate other initial states in redux, as we discussed.